### ISSUE TITLE
💡 [Guide] Checkstyle with Gradle

### ISSUE THREAD

**Issue by jetlfj** [nus-cs2103-AY2223S1/forum#35]
In case you are having difficulty adding Checkstyle to Gradle (like I did for quite a while), here is a quick guide:

After setting up Gradle in `A-Gradle`,
1. Create a directory `/config/checkstyle` within `ip`.
2. Copy and paste the files `checkstyle.xml` and `suppressions.xml` from [this repository](https://github.com/se-edu/addressbook-level3/tree/master/config/checkstyle) into this directory.
3. Within the `build.gradle` file, add the following:
```java
plugins {
    // other existing plugins
    id 'checkstyle'
}

checkstyle {
    // NOT '8.29'
    toolVersion = '10.2'
}
```
4. `gradlew checkstyleMain` or `./gradlew checkstyleMain` should work now.

**Comment by damithc** [comment]
Thanks for sharing this @jetlfj 
I've corrected the `toolVersion` as `10.2` in [this page](https://se-education.org/guides/tutorials/checkstyle.html#using-checkstyle-with-gradle).

EDIT: Also clarified about the two config files in the above page.

**Comment by lulucopter** [comment]
Just wanted to add on that I faced a similar issue, where the checkstyle file could not load with an exception along the lines of 

`The Checkstyle rules file could not be parsed. cannot initialize module TreeWalker - cannot initialize module JavadocMethod - Property 'scope' does not exist, please check the documentation` 

After some debugging, I realised the issue came from the addressbook file being outdated, since using the checkstyle version 8.32 rectifies the issue. If you face a similar issue, because you used the AB3-0.2.1.ea RAR file (Which was downloaded from the tP [releases page](https://github.com/se-edu/addressbook-level3/releases)), you may want to instead pull the files from the given repo (they should be the most updated at 10.2)
